src/main.rs

```diff
@@ -41,21 +59,48 @@ async fn main() {
 
     // Load settings
     let settings = if let Some(config_path) = &args.config {
-        Settings::load_from(config_path.into())
+        tracing::info!(
+            "Overrode default config path with provided '{}'",
+            config_path
+        );
+        Settings::load_from(config_path.into(), &args)
     } else {
-        Settings::load()
+        Settings::load(&args)
     };
 
-    let mut settings = match settings {
+    let settings = match settings {
         Ok(settings) => settings,
         Err(e) => {
             tracing::warn!("Failed to load settings, continuing with defaults: {:?}", e);
             Settings::default()
         }
     };
 
-    if let Some(port) = args.port {
-        settings.set_port(port);
+    let mut launch_opts = LaunchOptionsV2::new(
+        settings.get_steam_user().unwrap(),
+        gamefinder::TF2_GAME_ID.to_string(),
+    )
+    .context("Failed to extract launch options for main.")
```

Comment:
Not much use putting context on something you are just unwrapping immediately. Although it looks probably that this could fail so there should probably be some more graceful handling of the situation. We want to avoid `unwrap`s as much as possible, but if we are certain it won't fail and there is reason to have it then it should be an `expect` instead with a short description of what it was trying to do.